jose.py

```diff
@@ -529,7 +529,7 @@ def _jwe_hash_str(ciphertext, iv, adata='', legacy=False):
     # draft-ietf-jose-json-web-algorithms-24#section-5.2.2.1
     if legacy:
         return '.'.join((adata, iv, ciphertext, str(len(adata))))
-    return '.'.join((adata, iv, ciphertext, pack("!Q", len(adata) * 8)))
+    return ''.join((adata, iv, ciphertext, pack("!Q", len(adata) * 8)))
```

Comment:
Given there are now 3 different ways to compute this, solely for backwards compatibility, perhaps the `legacy` parameter should be changed to `version`, allowing for all three methods to persist until the legacy ones are dropped.